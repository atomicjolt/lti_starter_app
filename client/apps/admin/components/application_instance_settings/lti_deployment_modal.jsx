import React, { useState } from 'react';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';

import usePost from 'atomic-fuel/libs/hooks/use_post';

import Input from '../common/input';
import LtiInstallSelector from './lti_install_selector';


export default function LtiDeploymentModal(props) {
  const {
    isOpen,
    closeModal,
    applicationInstanceId,
    applicationId,
    setLastDeployment,
  } = props;

  const [ltiDeploymentId, setLtiDeploymentId] = useState('');
  const [ltiInstallId, setLtiInstallId] = useState();

  const {
    result, error, loading, postIt,
  } = usePost();

  function save() {
    const url = `api/application_instances/${applicationInstanceId}/lti_deployments`;
    postIt(url, {}, { deployment_id: ltiDeploymentId, lti_install_id: ltiInstallId });
  }

  let message = null;

  if (error) {
    message = error.message;
  } else if (loading) {
    message = 'Creating LTI Deployment...';
  } else if (result) {
    setLastDeployment();
    closeModal();
  }

  return (
    <ReactModal
      isOpen={isOpen}
      onRequestClose={() => closeModal()}
      contentLabel="Create LTI Deployment"
      overlayClassName="c-modal__background"
      className="c-modal c-modal--lti-deployment is-open"
    >
      <h2 className="c-modal__title">
        New LTI Deployment
      </h2>
      {message}
      <form>
        <div className="o-grid o-grid__modal-top">
          <div className="o-grid__item u-full">
            <Input
              className="c-input"
              labelText="Deployment ID"
              inputProps={{
                type: 'text',
                id: 'lti_deployment_id',
                value: ltiDeploymentId,
                onChange: (e) => setLtiDeploymentId(e.target.value),
              }}
            />
            <LtiInstallSelector
              applicationId={applicationId}
              ltiInstallId={ltiInstallId}
              setLtiInstallId={setLtiInstallId}
            />
          </div>
        </div>
        <button
          type="button"
          className="c-btn c-btn--yellow"
          onClick={save}
        >
          Save
        </button>
        <button
          type="button"
          className="c-btn c-btn--gray--large u-m-right"
          onClick={closeModal}
        >
          Cancel
        </button>
      </form>

    </ReactModal>
  );
}

LtiDeploymentModal.propTypes = {
  isOpen: PropTypes.bool,
  closeModal: PropTypes.func,
  applicationInstanceId: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]).isRequired,
  applicationId: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]).isRequired,
  setLastDeployment: PropTypes.func.isRequired,
};
