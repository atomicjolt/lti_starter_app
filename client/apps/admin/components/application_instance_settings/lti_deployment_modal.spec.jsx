import React from 'react';
import TestRenderer from 'react-test-renderer';

import { lti_deployment_modal } from './lti_deployment_modal';

describe('lti_deployment_modal', () => {
  it('renders the lti_deployment_modal component', () => {
    const result = TestRenderer.create(
        <lti_deployment_modal />
    );
    expect(result).toMatchSnapshot();
  });
});
