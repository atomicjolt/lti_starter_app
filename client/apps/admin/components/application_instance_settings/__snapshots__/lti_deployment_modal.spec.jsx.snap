// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lti_deployment_modal renders the lti_deployment_modal component 1`] = `<lti_deployment_modal />`;
